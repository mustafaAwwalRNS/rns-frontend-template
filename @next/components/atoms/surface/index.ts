export * from "./surface.types";
export * from "./surface.template";
