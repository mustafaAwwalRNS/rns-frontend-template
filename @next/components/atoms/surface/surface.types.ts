export interface SurfaceProps {
  children: React.ReactNode;
  spacing?: number;
}
