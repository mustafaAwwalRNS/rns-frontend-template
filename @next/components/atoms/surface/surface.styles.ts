import styled from "styled-components";
import { SurfaceProps } from "./surface.types";
export const Surface = styled.div<SurfaceProps>`
  ${({
    theme: {
      colors: { surface },
      spacing,
      borderRadius,
      shadow,
    },
    spacing: propSpacing,
  }) => ({
    backgroundColor: surface,
    borderRadius: borderRadius(1),
    padding: spacing(propSpacing || 1),
    boxShadow: shadow(2),
  })}
`;
