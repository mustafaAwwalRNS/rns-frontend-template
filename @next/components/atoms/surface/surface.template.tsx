import React from "react";
import { SurfaceProps } from "./surface.types";
import * as S from "./surface.styles";
export const Surface = ({ children, ...rest }: SurfaceProps): JSX.Element => {
  return <S.Surface {...rest}>{children}</S.Surface>;
};
