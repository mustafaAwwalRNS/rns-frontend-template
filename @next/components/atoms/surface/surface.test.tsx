import React from "react";
import { render, screen } from "test-utils";
import { Surface } from "./surface.template";
const text = "surface";
const setup = () => {
  render(<Surface>{text}</Surface>);
};

describe("<Surface />", () => {
  it("should render", () => {
    setup();
    expect(screen.getByText(text)).toBeInTheDocument();
  });
});
