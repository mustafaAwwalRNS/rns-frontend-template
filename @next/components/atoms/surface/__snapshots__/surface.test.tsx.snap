// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Surface /> should render 1`] = `
.c0 {
  background-color: white;
  border-radius: 0.5rem;
  padding: 0.5rem;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
}

<div>
  <div
    class="c0"
  >
    Surface content
  </div>
</div>
`;
